server/commitlog/message_set.go

```diff
@@ -48,7 +48,7 @@ func entriesForMessageSet(basePos int64, ms []byte) []*entry {
 	return entries
 }
 
-func newMessageSetFromProto(baseOffset, basePos int64, msgs []*Message) (
+func newMessageSetFromProto(baseOffset, basePos int64, msgs []*Message, concurrencyControl bool) (
```

Comment:
Would it make sense to panic here if `concurrencyControl` is `true` and `len(msgs) > 1`? This is a scenario that should not happen so might be good to enforce a sanity check.

Reply:
good point

WIP